app/shared/taginput/taginput.component.ts

```diff
@@ -0,0 +1,156 @@
+﻿import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { User } from '../../core/user/user.model';
+import { UserService } from '../../core/user/user.service';
+
+import {
+  Validators,
+  FormGroup,
+  FormControl
+} from '@angular/forms';
+
+@Component({
+  selector: 'hip-taginput',
+  templateUrl: './app/shared/taginput/taginput.component.html'
+})
+export class TagInputComponent {
+
+  public errorMessage: any;       // Handling error message
+  public names: string[] = [];    // AutoComplete List
+  @Input() role: string;          // User role Passed dynamically
+  @Input() users: number[];         // List of Users updated
```

Comment:
But maintaining consistency between arrays is far more easier(Part of taginput) than redundant code with duplicate logic!